### PR TITLE
Add optimizer state io

### PR DESCRIPTION
**Pull request by DogHeadMoon**
Add save/load state dict to make optimizers to save/load their states
Add learning_rate() to access learning rate value without waiting
Add a unit test file to test save/load the optimizers' states

### CODE REVIEW

**Review by iclementine — DISMISSED** [review]
LGTM！
